src/main/java/tracko/model/items/Price.java

```diff
@@ -0,0 +1,58 @@
+package tracko.model.items;
+
+import static tracko.commons.util.AppUtil.checkArgument;
+import static tracko.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+/**
+ *  Represents the price of an item.
+ */
+public class Price {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Price should be non-negative.";
+
+    public final BigDecimal price;
+
+    /**
+     * Constructs a {@code Price}.
+     * @param price The Price of an item.
+     */
+    public Price(BigDecimal price) {
+        requireAllNonNull(price);
+        checkArgument(isValidPrice(price), MESSAGE_CONSTRAINTS);
+        this.price = roundToNearestCent(price);
+    }
+
+    public static boolean isValidPrice(BigDecimal test) {
+        return !(test.compareTo(BigDecimal.ZERO) < 0);
+    }
```

Comment:
I think we can also implement checking Price validity to be a decimal with <= 2 decimal places here

Reply:
Oh I thought about that but I ended up just rounding off to the nearest cent internally Do you think if we already round it off (and perhaps state in the UG that all money will be rounded off to the nearest cent) that there will still be a need to validate the Price?